src/Form/Control/Password.php

```diff
@@ -4,7 +4,45 @@
 
 namespace Atk4\Ui\Form\Control;
 
+use Atk4\Ui\Button;
+use Atk4\Ui\Js\JsExpression;
+
 class Password extends Line
 {
     public string $inputType = 'password';
+
+    /** By setting this variable to true you get an Eye-button on the right side which can toggle the visibility of the entered password */
+    public bool $eye;
+
+    #[\Override]
+    protected function init(): void
+    {
+        parent::init();
+
+        if ($this->eye) {
+            $button = new Button([
+                'class.grey' => true,
+                'iconRight' => 'eye slash',
+            ]);
+
+            $this->action = $button;
+
+            $button->on(
+                'click',
+                new JsExpression(<<<EOF
+		    var input_element = document.querySelector(`[id$='form_layout_{$this->shortName}_input']`);
```

Comment:
using `shortName` to refering anything is wrong - see how it is done in other Form Controls

also use JS `let` instead of `var`

Reply:
I'm very unexperienced in ATK4 source-code writing, how would I go about to find the name of the input field I want to use in a Form Control? The only cases where it's used is:
Radio
Checkbox
Textarea

Reply:
see https://github.com/atk4/ui/blob/5.2.0/src/Form/Control/Input.php#L111 and https://github.com/atk4/ui/blob/5.2.0/src/Form/Control/Dropdown.php#L158

so the correct way to get the ID is `$this->name . '_input'`

it is guaranteed to be unique

Reply:
the 2nd template value must not use `shortName` too, as the ID can be based on shortened identifier, so the assumption with `shortName` is not guaranteed to be always true